pyanaconda/core/util.py

```diff
@@ -254,7 +254,7 @@ def sigusr1_preexec():
     finally:
         # Put everything back where it was
         signal.alarm(0)
-        signal.signal(signal.SIGUSR1, old_sigusr1_handler)
+        signal.signal(signal.SIGUSR1, signal.SIG_IGN)
```

Comment:
I am afraid that it is necessary to restore the previous signal handler.

Reply:
Oh. I forgot about that part. We use SIGUSR1 to test the crash handler. But if Xorg here keeps starting and eventually starts and sends it... goodbye Anaconda. So that's actually the original bug: Xorg sends the same signal that we use for debugging, and we take that to mean "let's crash".

Reply:
I guess we could try to kill `childproc` if we don't get X11 started in time? Timeout calls `sigalrm_handler`, so that would set some flag, and in `finally` we'd check the flag and try to kill the process.

(One alternative solution I can think of is: Ignore the first SIGUSR1 after trying to start X. But that introduces behavior that is very hard to guess when poking the program.)

Reply:
Can you be sure that the first SIGUSER1 signal received was sent by Xorg? I don't think it's quite right. And I don't quite understand, is there any other place in anaconda that uses this signal?

Reply:
In fact, I haven't found the code that sends SIGUER1. Could I please change the SIGUER1 sent by XORG to other signals?

Reply:
If I kill the Xorg process by calling terminate in finally block, is it possible that Xorg is calling IO and not responding to the signal

Reply:
From the man page:

```
       The X server attaches special meaning to the following signals:

       SIGHUP  This signal causes the server to close all  existing  connections,  free  all  re‐
               sources, and restore all defaults.  It is sent by the display manager whenever the
               main user's main application (usually an xterm or window manager) exits  to  force
               the server to clean up and prepare for the next user.

       SIGTERM This signal causes the server to exit cleanly.

       SIGUSR1 This  signal  is used quite differently from either of the above.  When the server
               starts, it checks to see if it has inherited SIGUSR1 as  SIG_IGN  instead  of  the
               usual SIG_DFL.  In this case, the server sends a SIGUSR1 to its parent process af‐
               ter it has set up the various connection schemes.  Xdm uses this feature to recog‐
               nize when connecting to the server is possible.
```

It doesn't look like it is possible to change SIGUSR1. However, it might be interesting to try SIGTERM to kill the X server.